examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp

```diff
@@ -403,21 +449,41 @@ CHIP_ERROR CastingServer::VerifyOrEstablishConnection(TargetVideoPlayerInfo & ta
         prevDeviceProxy->Disconnect();
     }
 
+    if (targetVideoPlayerInfo.IsAsleep())
+    {
+        ChipLogProgress(AppServer, "CastingServer::VerifyOrEstablishConnection(): Sending WoL to sleeping VideoPlayer and waiting");
+        ReturnErrorOnFailure(CastingServer::GetInstance()->SendWakeOnLan(targetVideoPlayerInfo));
+        usleep(CHIP_DEVICE_CONFIG_STR_WAKE_UP_DELAY_SEC * 1000 * 1000);
```

Comment:
what is the threading story if we sleep here?

any risks to any CX if we wait for this delay?

Reply:
At this point, the Matter thread will wait here - and it should because the tv-casting-app cannot do anything at this point (since it is not connected), but wait for the TV to turn on.

The WakeOnLAN magic packet follows a fire and forget paradigm, so we just wait for a certain amount of time, and then proceed with trying to connect. We can always fine tune the wait period as we do more testing.

Reply:
Shouldn't we ideally spin up another thread to do this then and have a callback? Would we want this to block other operations that might potentially come in?

Reply:
Done. We are now scheduling the verifyOrEstablishConnection call on the queue after CHIP\_DEVICE\_CONFIG\_STR\_WAKE\_UP\_DELAY\_SEC to avoid the blocking.